packages/components/src/toggle-group-control/toggle-group-control-option/component.tsx

```diff
@@ -1,96 +1,22 @@
-/**
- * External dependencies
- */
-import type { ForwardedRef } from 'react';
-// eslint-disable-next-line no-restricted-imports
-import { Radio } from 'reakit';
-
-/**
- * WordPress dependencies
- */
-import { useInstanceId } from '@wordpress/compose';
-
 /**
  * Internal dependencies
  */
-import {
-	contextConnect,
-	useContextSystem,
-	WordPressComponentProps,
-} from '../../ui/context';
-import type { ToggleGroupControlOptionProps, WithToolTipProps } from '../types';
-import { useToggleGroupControlContext } from '../context';
-import * as styles from './styles';
-import { useCx } from '../../utils/hooks';
-import Tooltip from '../../tooltip';
-
-const { ButtonContentView, LabelView } = styles;
+import type { WordPressComponentProps } from '../../ui/context';
+import type { ToggleGroupControlOptionProps } from '../types';
+import { ToggleGroupControlOptionBase } from '../toggle-group-control-option-base';
 
-const WithToolTip = ( { showTooltip, text, children }: WithToolTipProps ) => {
-	if ( showTooltip && text ) {
-		return (
-			<Tooltip text={ text } position="top center">
-				{ children }
-			</Tooltip>
-		);
-	}
-	return <>{ children }</>;
-};
-
-function ToggleGroupControlOption(
-	props: WordPressComponentProps< ToggleGroupControlOptionProps, 'button' >,
-	forwardedRef: ForwardedRef< any >
+export default function ToggleGroupControlOption(
+	props: WordPressComponentProps< ToggleGroupControlOptionProps, 'button' >
```

Comment:
We should mark `ToggleGroupControlOptionBase`, `ToggleGroupControlOption` and `ToggleGroupControlOptionIcon` as non-polymorphic, since they are internally always going to render `as="button"`

```suggestion
	props: WordPressComponentProps< ToggleGroupControlOptionProps, 'button', false >
```